app/facades/user_books_facade.rb

```diff
@@ -1,7 +1,32 @@
 class UserBooksFacade
 
-  def self.all_books
-    
+  def self.all_books(auth_token)
+    shelves = BookService.book_shelves(auth_token)
+    shelf_ids = shelves[:items].map do |shelf|
+      if shelf[:title] == "Reading now" || shelf[:title] == "To read" || shelf[:title] == "Have read"
+        { id: shelf[:id], title: shelf[:title] }
+      end
+    end.compact
+    all_books = shelf_ids.map do |shelf|
+      contents = BookService.books_on_shelf(shelf[:id], auth_token)[:items]
+      contents.map do |book|
+        BookAndShelf.new(
+          g_id: book[:id],
+          title: book[:volumeInfo][:title],
+          authors: book[:volumeInfo][:authors],
+          description: book[:volumeInfo][:description],
+          genres: book[:volumeInfo][:categories],
+          shelves: [shelf[:title]]
+        )
+      end if contents
```

Comment:
Is this `if contents` to accommodate cases when `BookService.books_on_shelf()` returns an empty array?